python/xir/parser.py

```diff
@@ -183,26 +189,42 @@ def statement(self, args):
     def gatestmt(self, args):
         """Gate statements that are defined directly in the circuit or inside
         a gate declaration."""
+
+        adjoint = False
+        ctrl_wires = tuple()
```

Comment:
If we're planning to modify `ctrl_wires`, we should make it a list (and later cast it to a tuple if necessary).
```suggestion
        ctrl_wires = []
```